### PR TITLE
[WIP] [Reader] #1983 Reduce wasted renders while searching

### PR DESCRIPTION
**Pull request by NickHeiner**
Connect #1983.

This actually is not making a huge perf difference, but I found that I could make some small improvements while I was researching what was going on, so I figured we might as well merge them.

The biggest change is using `PureComponent` as often as possible.

### Wasted Renders – Before
image

### Wasted Renders – After
image

### CODE REVIEW

File: client/app/components/DocumentCategoryIcons.jsx [hunk 1]
```diff
@@ -4,40 +4,42 @@ import { connect } from 'react-redux';
...
-};
+            {category.svg}
+          </li>
+        )
+      }
+    </ul>;
+  }
+}
+
+const mapPropsToState = (state, ownProps) => ({
```
Comment (mdbenjam) [inline client/app/components/DocumentCategoryIcons.jsx]: Should be mapStateToProps

File: client/app/components/DocumentCategoryIcons.jsx [hunk 2]
```diff
@@ -4,40 +4,42 @@ import { connect } from 'react-redux';
...
-};
+            {category.svg}
+          </li>
+        )
+      }
+    </ul>;
+  }
+}
+
+const mapStateToProps = (state, ownProps) => ({
```
Comment (NickHeiner) [inline client/app/components/DocumentCategoryIcons.jsx]: For this component, we save some renders by moving logic from `render` to `mapStateToProps`. 

If we see props move from `{doc: Document(id=1)}` to `{doc: Document(id=2)}`, that would trigger a re-render. But what we really care about is the categories for each document (eg `{medical: true, procedural: false}`), and if those are the same between two docs, we don't need to re-render.

File: lib/fakes/appeal_repository.rb [hunk 3]
```diff
@@ -350,14 +350,15 @@ def self.static_reader_documents
     ]
   end
 
-  def self.random_reader_documents(num_documents)
+  def self.random_reader_documents(num_documents, seed = Random::DEFAULT.seed)
+    seededRandom = Random.new(seed)
```
Comment (mdbenjam) [inline lib/fakes/appeal_repository.rb]: I like this change a lot 👍

Comment (NickHeiner) [inline lib/fakes/appeal_repository.rb]: I used a seeded Random here so I would get consistent results between page reloads. It freaked me out when I'd get different results for the same search queries after making a code change. This eliminates the possibility that the difference is due to a change I made, and not randomness, but I think that it's randomness and thus miss a bug.

**Pull request merged**